### ISSUE TITLE
Parse inlineLink incorrectly if the link text is an URL

### ISSUE THREAD

**Issue by hoducha** [erusev/parsedown#358]
Test data:

```
[http://example.com](http://example.com)
```

Expected result:

``` html
<a href="http://example.com">http://example.com</a>
```

Actual result:

``` html
<a href="http://example.com"><a href="http://example.com">http://example.com</a></a>
```

**Comment by petermolnar** [comment]
Duplicate of https://github.com/erusev/parsedown-extra/issues/77 ?

**Comment by petermolnar** [comment]
This could be the reason behind https://github.com/erusev/parsedown-extra/issues/77

**Comment by francisbesset** [comment]
@petermolnar This issue can be reproduced with parsedown only.

**Comment by aidantwoods** [comment]
I've added made a pull request https://github.com/erusev/parsedown/pull/433 to address this issue

**Comment by aidantwoods** [comment]
Fixed in #433
